### PR TITLE
Fix panel title colours and focus

### LINKED ISSUE #6815: All panel & chevrons should change colour when expanded
## Bug Description

originally reported by @ivayloklissarov 
The Title and chevron state should turn to Primary Grey once expanded.

## Screenshots
image2

## Additional Context

- Plugin Version:
- WordPress Version:
- Operating System:
- Browser:

---

_Do not alter or remove anything below. The following sections will be managed by moderators only._

### PR DESCRIPTION
**Pull request by miina**
## Context
With this PR the panel title colours differ when collapsed/expanded.

## User-facing changes
| Before | After |
|--------|------|
| before | title-after |
| Screenshot 2021-03-18 at 09 47 06 | Screenshot 2021-03-18 at 09 44 59 |

### QA

- [ ] This is a non-user-facing change and requires no QA

This PR can be tested by following these steps:

1. Expand and collapse the design panels.
2. Verify the title colour changes to lighter when expanded and to darker when collapsed.
3. Tab into a panel header and verify the focus line around the headline looks as expected, too.

See [Figma.](https://www.figma.com/file/bMhG3KyrJF8vIAODgmbeqT/Design-System?node-id=3389%3A38563)

### UAT

- [x] UAT should use the same steps as above.

This PR can be tested by following these steps:

1.

## Checklist

- [x] This PR addresses an existing issue and I have linked this PR to it in ZenHub
- [x] I have tested this code to the best of my abilities
- [x] I have verified accessibility to the best of my abilities ([docs](https://github.com/google/web-stories-wp/blob/main/docs/accessibility-testiing.md))
- [x] I have verified i18n and l10n (translation, right-to-left layout) to the best of my abilities
- [ ] This PR contains automated tests (unit, integration, and/or e2e) to verify the code works as intended ([docs](https://github.com/google/web-stories-wp/tree/main/docs#testing))
- [x] I have added documentation where necessary
- [x] I have added a matching `Type: XYZ` label to the PR

---

Fixes #6815

### CODE REVIEW

File: assets/src/edit-story/components/panels/panel/shared/title.js [hunk 1]
```diff
@@ -32,8 +32,10 @@ import {
   Icons,
   THEME_CONSTANTS,
   Headline,
+  themeHelpers,
+  ThemeGlobals,
 } from '../../../../../design-system';
-import { KEYBOARD_USER_SELECTOR } from '../../../../utils/keyboardOnlyOutline';
+import { BUTTON_TRANSITION_TIMING } from '../../../../../design-system/components/button/constants';
```
Comment (miina) [inline assets/src/edit-story/components/panels/panel/shared/title.js]: Should we move this to general constants in the design system perhaps?

Comment (samwhale) [reply assets/src/edit-story/components/panels/panel/shared/title.js]: Yeah this is an awkward import 😅, I changed this in a different pr too. I ended up doing this: https://github.com/google/web-stories-wp/pull/6786/files#diff-bc0076673501c056995205221a976b39f5fb74b1febffaa3a8f9c5195356eb60R18

File: assets/src/edit-story/components/panels/panel/shared/title.js [hunk 2]
```diff
@@ -74,20 +76,27 @@ const IconWrapper = styled.div`
...
   padding: 0 4px 0 0;
   cursor: pointer;
+  margin-left: -12px;
+  transition: ${BUTTON_TRANSITION_TIMING};
+
+  &:hover,
+  &.${ThemeGlobals.FOCUS_VISIBLE_SELECTOR}, &[data-focus-visible-added] {
+    color: ${({ theme }) => theme.colors.fg.primary};
+  }
+  ${themeHelpers.focusableOutlineCSS};
```
Comment (samwhale) [inline assets/src/edit-story/components/panels/panel/shared/title.js]: This looks so good. The focus styling is layered box shadows, but the inner box shadow defaults to the primary background color. If we want the inner black border to match the panel background we could do:
```
${({ theme }) => 
    themeHelpers.focusableOutlineCSS(
        theme.colors.border.focus, 
        theme.colors.bg.secondary
    )};
```

We haven't figured out a way to have the focus styles inherit the color for the inner box shadow yet 😭. 

image

File: assets/src/edit-story/components/panels/panel/shared/title.js [hunk 3]
```diff
@@ -74,20 +76,27 @@ const IconWrapper = styled.div`
...
+    isCollapsed ? theme.colors.fg.secondary : theme.colors.fg.primary};
   height: 32px;
   display: flex; /* removes implicit line-height padding from child element */
   padding: 0 4px 0 0;
   cursor: pointer;
+  margin-left: -12px;
+  transition: ${BUTTON_TRANSITION_TIMING};
+
+  &:hover,
+  &.${ThemeGlobals.FOCUS_VISIBLE_SELECTOR}, &[data-focus-visible-added] {
```
Comment (samwhale) [inline assets/src/edit-story/components/panels/panel/shared/title.js]: Oh we have a constant for this too!

```suggestion
  &.${ThemeGlobals.FOCUS_VISIBLE_SELECTOR}, &[${ThemeGlobals.FOCUS_VISIBLE_DATA_ATTRIBUTE}] {
```

**Review by samwhale — APPROVED** [review]
I love the new styling

File: assets/src/edit-story/components/panels/panel/shared/title.js (hunk 2 above)
Comment (miina) [reply assets/src/edit-story/components/panels/panel/shared/title.js]: Hm, it would ideally be transparent + only the blue line so it would work on top of any background. Perhaps that's something that could have its own issue for adjusting this from the design-system directly.

**Review by barklund — APPROVED** [review]
Looks good pending @samwhale's comments.

File: assets/src/edit-story/components/panels/panel/shared/title.js (hunk 2 above)
Comment (BrittanyIRL) [reply assets/src/edit-story/components/panels/panel/shared/title.js]: if we need it to be 'transparent' there might be a use case for a different focusableOutline shared style that isn't centered on traditional buttons/focus content. Since this is getting applied directly on the panel title it seems like a good use case for say, not having the break that would need the "transparent" since it doesn't do anything for the focus style to have that break. 

tl;dr is that with the focus style as it is there _must_ be a color given for that first value, this has been the topic of some lengthy discussions with sam and the color between content and focus-blue border is a known trade off.  Just sayin', instead of reworking that we might as well have focus variant for situations like this that don't benefit at all from the extra 2px space. 

thoughts?

**Review by BrittanyIRL — APPROVED** [review]
thank you for fixing this isolated!

File: assets/src/edit-story/components/panels/panel/shared/title.js (hunk 2 above)
Comment (samwhale) [reply assets/src/edit-story/components/panels/panel/shared/title.js]: Would be so great to have it be transparent, however if it's transparent then it'll show the blue box shadow that's underneath it 😅 

We're really using the two box shadows to show an illusion that there's a border 2px away from the content but they both begin at the content edge.

Comment (miina) [reply assets/src/edit-story/components/panels/panel/shared/title.js]: Thanks for the context, makes sense. Yep, if that's something that's not an easy change then we could potentially have a variant or perhaps just accept `backgroundColor`.

I'll just change this to the correct colour for now 👍 

Could we change the default colour for `darkTheme` to `theme.colors.bg.secondary`? It looks like the focusing is mostly used on the panels anyway. Ah -- I suppose that would change it for the `light` theme as well though. Hm, what if we create a new color code for the themes (e.g. something like `focusHelperLine`, surely there's a better name 😛 ) which for the editor would be `theme.colors.bg.secondary` and for the DB `theme.colors.bg.primary` and use this inside `focusableOutlineCSS` instead? Thoughts?

File: assets/src/edit-story/components/panels/panel/shared/title.js (hunk 1 above)
Comment (miina) [reply assets/src/edit-story/components/panels/panel/shared/title.js]: It looks like `export * from './button';` is exporting constants that are also exported by `export * from './constants';` so there would actually be a collision of exports this way. Maybe we should export it from the `/button` directly instead. Thoughts?

File: assets/src/edit-story/components/panels/panel/shared/title.js (hunk 2 above)
Comment (BrittanyIRL) [reply assets/src/edit-story/components/panels/panel/shared/title.js]: i think we're gonna have issues with any of these! I vote we table it until we can see all of the use cases for it and then make a choice on how to clean this up separately with broader context. I still think we might benefit from a version that just cuts out the "transparent" (when i say transparent i just mean from a design perspective) for things like this because the space isn't even a think we need on these cases.

File: assets/src/edit-story/components/panels/panel/shared/title.js (hunk 1 above)
Comment (BrittanyIRL) [reply assets/src/edit-story/components/panels/panel/shared/title.js]: oh no! yeah i guess exporting from button is the best way to do this then.

File: assets/src/edit-story/components/panels/panel/shared/title.js (hunk 2 above)
Comment (miina) [reply assets/src/edit-story/components/panels/panel/shared/title.js]: Alright, sgtm!

File: assets/src/edit-story/components/panels/panel/shared/title.js (hunk 1 above)
Comment (samwhale) [reply assets/src/edit-story/components/panels/panel/shared/title.js]: Up to you! we'd have to change it in a few places for it to work:
image

My preference is to export constants from their own file but I think either is fine.
